Zom/Zom/Classes/View Controllers/ZomFingerprintBaseViewController.swift

```diff
@@ -0,0 +1,220 @@
+     before `self.viewDidLoad()` was called!
+
+     - parameter buddy: a OTRBuddy with a new OMEMO or OTR fingerprint.
+    */
+    @objc convenience init(buddy: OTRBuddy?) {
+        self.init()
+        self.buddy = buddy
+    }
+
+    /**
+     Convenience initializer which also sets `self.buddy`, `self.omemoDevices` and
+     `self.otrFingerprints` with the given arguments.
+
+     You can do this yourself, since all of these are internal public, but you need to do this,
+     before `self.viewDidLoad()` was called!
+
+     - parameter buddy: a OTRBuddy with a new OMEMO or OTR fingerprint.
+     - parameter omemoDevices: a list of OMEMODevices of that buddy.
+     - parameter otrFingerprints: a list of OTRFingerprints of that buddy.
+    */
+    @objc convenience init(buddy: OTRBuddy?, omemoDevices: [OMEMODevice]?, otrFingerprints: [OTRFingerprint]?) {
+        self.init()
+        self.buddy = buddy
+        self.omemoDevices = omemoDevices ?? []
+        self.otrFingerprints = otrFingerprints ?? []
+    }
+
+    /**
+     Load all `OMEMODevice`s and `OTRFingerprint`s of the set `OTRBuddy`, *if* they are not set,
+     yet.
+    */
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        if omemoDevices.count < 1 && otrFingerprints.count < 1,
+            let buddy = buddy, let db = OTRDatabaseManager.sharedInstance().readConnection {
+            db.asyncRead() { (transaction) in
+                self.omemoDevices = OMEMODevice.allDevices(
+                    forParentKey: buddy.uniqueId,
+                    collection: type(of: buddy).collection,
+                    transaction: transaction)
+                    .filter() { (device) -> Bool in
+                        return device.publicIdentityKeyData != nil && device.trustLevel != .removed
+                    }
+
+                if let account = buddy.account(with: transaction) {
+                    self.otrFingerprints = OTRProtocolManager.encryptionManager.otrKit.fingerprints(
+                        forUsername: buddy.username,
+                        accountName: account.username,
+                        protocol: account.protocolTypeString())
+                }
+
+                DispatchQueue.main.async {
+                    self.fingerprintsLoaded()
+                }
+            }
+        }
+        else {
+            fingerprintsLoaded()
+        }
+
+        let titleView = OTRTitleSubtitleView(frame: CGRect(x: 0, y: 0, width: 200, height: 40))
+        titleView.titleLabel.textColor = UIColor.white
+        titleView.subtitleLabel.textColor = UIColor.white
+        titleView.subtitleLabel.text = buddy?.username
+        navigationItem.titleView = titleView
+
+        navigationItem.rightBarButtonItem = UIBarButtonItem.init(
+            title: CANCEL_STRING(), style: .done, target: self, action: #selector(cancel))
+
+        avatarImg.image = buddy?.avatarImage
+
+        badgeLb.font = UIFont(name: "icomoon", size: badgeLb.font.pointSize)
```

Comment:
Any reason why you set the font in code? I think it would be better to set them in the xib (font -> Custom, family -> icomoon). Minor thing, but I noticed this in some places in the code.